### ISSUE TITLE
MissingPluginException(No implementation found..) kotlin.

### ISSUE THREAD

**Issue by joshan39** [BlinkID/blinkid-flutter#5]

**Comment by matvidako** [comment]
Hi,

can you explain your issue in more detail?

**Comment by joshan39** [comment]
Microblink is already generated in flutter android project as 
 flutterEngine.getPlugins().add(new com.microblink.flutter.MicroblinkFlutterPlugin());    //in GeneratedPluginRegistrant

and in MainActivity: FlutterFragmentActivity
GeneratedPluginRegistrant.registerWith(flutterEngine); 

will resolved all the dependency.

However, in the blinkid-flutter documentation

Android
Android folder is fully initialized after running ./initFlutterDemo.sh. The script will replace default MainActivity.java with the ./sample_files/MainActivity.java. This is important, because the plugin works only if MainActivity extends io.flutter.app.FlutterActivity and if it is registered in the following way MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin")).

this last line I was unable to put . which is causing the Missing plugin exception .

And there is no work around.

**Comment by matvidako** [comment]
Hi,

you can try removing 
```
flutterEngine.getPlugins().add(new com.microblink.flutter.MicroblinkFlutterPlugin());
GeneratedPluginRegistrant.registerWith(flutterEngine);
```
and only using 
`MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin")).
`
You can use the same approach for other plugins, as described [here](https://github.com/BlinkID/blinkid-flutter/issues/2).

**Comment by joshan39** [comment]
Do you mind suggesting this for kotlin:

import android.os.Bundle;
import io.flutter.app.FlutterActivity;
import com.microblink.flutter.MicroblinkFlutterPlugin;
import io.flutter.plugins.urllauncher.UrlLauncherPlugin;

public class MainActivity extends FlutterActivity {
  @Override
  public void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);

    MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin"));
    UrlLauncherPlugin.registerWith(registrarFor("io.flutter.plugins.urllauncher.UrlLauncherPlugin"));
  }
}

**Comment by matvidako** [comment]
Hi, 

what kind of error do you get?
Kotlin isn't much different
```
import android.os.Bundle
import com.microblink.flutter.MicroblinkFlutterPlugin
import io.flutter.app.FlutterActivity
import io.flutter.plugins.urllauncher.UrlLauncherPlugin

class MainActivity : FlutterActivity() {
    public override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin"))
        UrlLauncherPlugin.registerWith(registrarFor("io.flutter.plugins.urllauncher.UrlLauncherPlugin"))
    }
}
```

**Comment by joshan39** [comment]
class MainActivity: FlutterFragmentActivity() {
    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
        GeneratedPluginRegistrant.registerWith(flutterEngine);
    }
}
I dont have bundle to work with and using flutter version 1.17.5.

**Comment by dCubelic** [comment]
Hi @joshan39 ,

could you please describe the issues you are experiencing on iOS in more detail?

**Comment by joshan39** [comment]
As above both are experiencing MissingPluginException(No implementation found..)

**Comment by dCubelic** [comment]
Hi @joshan39 ,

could you try running `flutter clean` and then `flutter pub get`. After that try deleting the app from the simulator / device.
If that does not work, could you please provide us with a minimal sample app that reproduces the issue so we can take a better look into the problem.

Best regards,
Dominik

**Comment by joshan39** [comment]
I have created the sample project replicating the issue above. Please give it a check.
https://github.com/JoshanTandukar/flutter_app

**Comment by codeguesser** [comment]
1.I copied BlinkID to path ../BlinkID folder
2.blinkid_flutter:
    path: ../BlinkID
copy them to pubspec.yaml
3. flutter pub get
4. flutter -d emulator-5554 run
5. it print error:
 Unhandled Exception: MissingPluginException(No implementation found for method scanWithCamera on channel microblink_scanner)

can you help me?
@joshan39

**Comment by dCubelic** [comment]
@joshan39 I downloaded your sample, changed the license and added the Camera usage string to info.plist and the app loads and scans my ID fine, with no errors.

**Comment by joshan39** [comment]
Thanks, but android still having issue.
@matvidako I am still having the issue in Android.
I will give it a check in IOS

**Comment by joshan39** [comment]
And Also I cannot see the license field for flutter . What should I do ? Generate separated license for android and IOS?

**Comment by SandraKuzmic** [comment]
Hi @joshan39 

as far as I can see in your sample, there is no `io.flutter.plugins.GeneratedPluginRegistrant`file and MainActivity is empty. GeneratedPluginRegistrant should be generated automatically while MainActivity should be similar to [this](https://github.com/BlinkID/blinkid-flutter/blob/master/sample_files/MainActivity.java).

As of the license, yes you have to generate separated licenses for android and iOS as stated [here](https://github.com/BlinkID/blinkid-flutter#plugin-usage).

**Comment by AHEqab** [comment]
I'm having the same issue. Please provide us the code required for Kotlin

**Comment by SandraKuzmic** [comment]
Hi @AHEqab 

here is required code for MainActivity in Kotlin
```
// TODO add your package name here

import android.os.Bundle
import com.microblink.flutter.MicroblinkFlutterPlugin
import io.flutter.app.FlutterActivity

class MainActivity : FlutterActivity() {
    public override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin"))
    }
}
```

Hope it helps,
Sandra

**Comment by AHEqab** [comment]
Hi @SandraKuzmic ,
This will not work

**Comment by SandraKuzmic** [comment]
Hi @AHEqab 

can you provide more information on the issue you are having? Which error are you getting?

If it is possible it would help us if you could send minimal sample app so we can reproduce the issue.

Best regards,
Sandra

**Comment by AHEqab** [comment]
Hi @SandraKuzmic ,

kindly find the below snippets of the MainActivity:

`import android.os.Bundle
import androidx.annotation.NonNull;
import io.flutter.embedding.android.FlutterActivity
import io.flutter.embedding.engine.FlutterEngine
import io.flutter.plugins.GeneratedPluginRegistrant

class MainActivity: FlutterActivity() {
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin"))
    }
    override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
        GeneratedPluginRegistrant.registerWith(flutterEngine);
    }
}
`

I'm getting the following error:

`Running Gradle task 'assembleDebug'...
e: _/path/to/project_/MainActivity.kt: (12, 9): Unresolved reference: MicroblinkFlutterPlugin
e: _/path/to/project_/MainActivity.kt: (12, 46): Unresolved reference: registrarFor

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:compileDebugKotlin'.
> Compilation error. See log for more details

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 11s
Exception: Gradle task assembleDebug failed with exit code 1`

The Main Activity usually is not modified manually in flutter for plugins

**Comment by SandraKuzmic** [comment]
Hi @AHEqab 

your MainActivity should extend `io.flutter.app.FlutterActivity` and not `io.flutter.embedding.android.FlutterActivity` so you need to change the import line in order for it to work.

Also, you can delete this
```
override fun configureFlutterEngine(@nonnull flutterEngine: FlutterEngine) {
GeneratedPluginRegistrant.registerWith(flutterEngine);
}
```

since the plugin is registered in the onCreate method (code from the snippet I sent in the previous message).

Best regards,
Sandra

**Comment by AHEqab** [comment]
Hi @SandraKuzmic 

I got it to compile using the above. However, using this approach and by removing the configureFlutterEngine function, all my other plugins does not work.

**Comment by SandraKuzmic** [comment]
Hi @AHEqab 

you should be able to add all the other plugins in the same way as BlinkID plugin is added. Simply add the desired plugins in onCreate method, something like this...

```
import android.os.Bundle
import com.microblink.flutter.MicroblinkFlutterPlugin
import io.flutter.app.FlutterActivity
import io.flutter.plugins.urllauncher.UrlLauncherPlugin

class MainActivity : FlutterActivity() {
    public override fun onCreate(savedInstanceState: Bundle) {
        super.onCreate(savedInstanceState)
        MicroblinkFlutterPlugin.registerWith(registrarFor("com.microblink.flutter.MicroblinkFlutterPlugin"))
        UrlLauncherPlugin.registerWith(registrarFor("io.flutter.plugins.urllauncher.UrlLauncherPlugin"))
        // or generally: PluginName.registerWith(registrarFor("your.plugin.package.name.PluginName"))
    }
}
```

I used `UrlLauncherPlugin` for this example but you can replace it with your plugins, just make sure to add the correct package name.

Best regards,
Sandra

**Comment by joshan39** [comment]
Please add support for FlutterEngine.

**Comment by SandraKuzmic** [comment]
Hi,

we added support for FlutterEngine and it is now the default way of registering BlinkID plugin.

The fix is released as part of the [v5.7.0 release](https://github.com/BlinkID/blinkid-flutter/releases/tag/v5.7.0).

Best regards,
Sandra

**Comment by anjapenic** [comment]
Hello all,

Since this was fixed in v5.7.0, we're closing the issue. If you need any further assistance, feel free to contact us at support@microblink.com or open a new issue.

Best regards,
Anja
